packages/keyring-controller/src/KeyringController.ts

```diff
@@ -658,18 +674,27 @@ export class KeyringController extends BaseController<
    * Adds a new account to the default (first) HD seed phrase keyring.
    *
    * @param accountCount - Number of accounts before adding a new one, used to
+   * @param keyringId - The id of the keyring to add the account to.
    * make the method idempotent.
    * @returns Promise resolving to the added account address.
    */
-  async addNewAccount(accountCount?: number): Promise<string> {
+  async addNewAccount(
+    accountCount?: number,
+    keyringId?: string,
+  ): Promise<string> {
     return this.#persistOrRollback(async () => {
-      const primaryKeyring = this.getKeyringsByType('HD Key Tree')[0] as
-        | EthKeyring<Json>
-        | undefined;
-      if (!primaryKeyring) {
+      let selectedKeyring: EthKeyring<Json> | undefined;
+      if (keyringId) {
+        selectedKeyring = this.getKeyringById(keyringId) as EthKeyring<Json>;
+      } else {
+        selectedKeyring = this.getKeyringsByType(
+          'HD Key Tree',
+        )[0] as EthKeyring<Json>;
+      }
+      if (!selectedKeyring) {
```

Comment:
Calling `this.withKeyring` internally here would avoid extra complexity and would delegate to withKeyring the following responsibilities:

1. Keyring selection
2. Safeguarding the operation with controller locks
3. Handle the case where the keyring is inexistent